test/app/components/market/submit/market-application.model.spec.ts

```diff
@@ -0,0 +1,137 @@
+            type: NcjParameterRawType.string,
+            metadata : {
+                description: "Param Description",
+            },
+            defaultValue: "jobname",
+        });
+        expect(parameter.type).toBe(NcjParameterExtendedType.string);
+    });
+
+    it("should have default", () => {
+        parameter = new NcjParameterWrapper("jobName", {
+            type: NcjParameterRawType.string,
+            metadata : {
+                description: "Param Description",
+            },
+            defaultValue: "jobname",
+        });
+        expect(parameter.defaultValue).toBe("jobname");
+    });
+
+    it("should not have default", () => {
+        parameter = new NcjParameterWrapper("jobName", {
+            type: NcjParameterRawType.string,
+            metadata : {
+                description: "Param Description",
+            },
+        });
+        expect(parameter.defaultValue).toBeUndefined();
+    });
+
+    it("should have allowedValues", () => {
+        parameter = new NcjParameterWrapper("jobName", {
+            type: NcjParameterRawType.string,
+            metadata : {
+                description: "Param Description",
+            },
+            allowedValues : ["a", "b", "c", "d"],
+        });
+        expect(parameter.allowedValues.length).toBe(4);
+    });
+
+    it("should not have allowedValues", () => {
+        parameter = new NcjParameterWrapper("jobName", {
+            type: NcjParameterRawType.string,
+            metadata : {
+                description: "Param Description",
+            },
+        });
+        expect(parameter.allowedValues.length).toBe(0);
+    });
+
+    it("should have description", () => {
+        parameter = new NcjParameterWrapper("jobName", {
+            type: NcjParameterRawType.string,
+            metadata : {
+                description: "description",
+            },
+        });
+        expect(parameter.description).toBe("description");
+    });
+
+    it("should have dependsOn", () => {
+        parameter = new NcjParameterWrapper("jobName", {
+            type: NcjParameterRawType.string,
+            metadata : {
+                description: "description",
+                dependsOn: "dependson",
+            },
+        });
+        expect(parameter.dependsOn).toBe("dependson");
+    });
+
+    it("should not have dependsOn", () => {
+        parameter = new NcjParameterWrapper("jobName", {
+            type: NcjParameterRawType.string,
+            metadata : {
+                description: "description",
+            },
+        });
+        expect(parameter.dependsOn).toBeUndefined();
+    });
+
+    it("should detect int type", () => {
+        parameter = new NcjParameterWrapper("frameEnd", {
+            type: NcjParameterRawType.int,
+            metadata : {
+                description: "description",
+            },
+            defaultValue: 4,
+        });
+        expect(parameter.type).toBe(NcjParameterExtendedType.int);
+    });
```

Comment:
May want to add a unit test here to check for the value as well (since this is an int which is a different case than what you have above for string).

Reply:
Same goes for cases below.

Reply:
done